BankClient/Bank.cs

```diff
@@ -2,165 +2,204 @@
-        Console.WriteLine("---------Menu ends---------");
-        Console.Write("Choose action: ");
+        new UserInformation("Dmitry", "Lapshin", "1234")
+            {ID = Guid.NewGuid()},
+        new UserInformation("Dima", "Lapshin", "1234")
+            {ID = Guid.NewGuid()}
+    };
+    /// <summary>
+    /// Get information from user
+    /// </summary>
+    /// <returns>Nothing</returns>
+    public UserInformation GetInformationFromUser()
+    {
+        Console.WriteLine("Firstname: ");
+        var firstName = Console.ReadLine();
+        firstName.Trim();
+        Console.WriteLine("Lastname: ");
+        var lastName = Console.ReadLine();
+        lastName.Trim();
+        Console.WriteLine("PhoneNumber: ");
+        var phoneNumber = Console.ReadLine();
+        phoneNumber.Trim();
+        var saveDataToClass = new UserInformation(firstName, lastName, phoneNumber);
+        return saveDataToClass;
+    }
+    /// <summary>
+    /// Function filters list of users which appropriate to inputUser
+    /// </summary>
+    /// <param name="inputUser">Takes information about client which user input</param>
+    /// <returns>List of users which appropriate to input user</returns>
+    public List<UserInformation> SearchUsersInformation(UserInformation inputUser)
+    {
+        var filterList = Users.Where(listUsers =>
+            (listUsers.FirstName == inputUser.FirstName || inputUser.FirstName == "") &&
+            (listUsers.LastName == inputUser.LastName || inputUser.LastName == "") &&
+            (listUsers.PhoneNumber == inputUser.PhoneNumber || inputUser.PhoneNumber == "")).ToList();
+        return filterList;
     }
 
-    private List<UserInformation> SearchUsersInformation()
+    /// <summary>
+    /// Function saves filterList and call function for print
+    /// </summary>
+    /// <param name="filterList"></param>
+    /// <returns>groupOfUsers</returns>
+    public List<UserInformation> GetAllUsersInformation(List<UserInformation> filterList)
     {
-        Console.WriteLine("Example:Dmitry,Lapshin,0950095521");
-        Console.Write("Fiter: ");
-        var userfilter = Console.ReadLine().Split(',');
-        foreach (var param in userfilter)
+        var groupOfUsers = filterList;
+        
+        if (groupOfUsers != null)
         {
-            param.Trim();
+            
+            ShowListOfUsers(groupOfUsers);
         }
 
-        if (userfilter[0] == "" && userfilter.Length == 1)
+        return groupOfUsers;
+        
+    }
+
+    /// <summary>
+    /// Gets one user
+    /// </summary>
+    /// <param name="filterList">Takes <see cref="UserInformation"/> list which contains all user information</param>
+    /// <returns>Nothing</returns>
+
+    public UserInformation GetOneUserInformation(List<UserInformation> filterList)
```

Comment:
Rename it to `GetUserInformation`. We need only one object to filter - use `UserInformation` as argument

Reply:
Ok